### PR TITLE
chore: should not perform diff and clone operations on evalProps

### LINKED ISSUE #31092: Remove dependency of evalProps in diff and clone computations.
We have to not perform diff and clone computations of evalProps in the worker thread and subsequently ensure the application continues to work the same. Through this step we should expect a significant improvement in performance.

### PR DESCRIPTION
**Pull request by vsvamsi1**
## Description
We have to not perform diff and clone computations of evalProps in the worker thread and subsequently ensure the application continues to work the same. Through this step we should expect a significant improvement in performance.

#### PR fixes following issue(s)
Fixes #31092

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
	- Optimized data handling and evaluation processes across various components.
- **New Features**
	- Enhanced the `CodeEditor` component to support additional validation scenarios for dynamic properties.
- **Bug Fixes**
	- Removed unnecessary dependencies and streamlined data retrieval in property controls.

### CODE REVIEW

File: app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx
Comment (rishabhrathod01) [inline app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx:1]: `PrimaryColumnsControl.tsx` and `PrimaryColumnControlV2.tsx` also has the `getPropertyValidation` method and these are with `isTriggerPath` check, do we not need to add the check in these files too same as `CodeEditor/index.tsx`

Comment (vsvamsi1) [reply app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx:1]: PrimaryColumns do not support click actions, so they cannot be a triggerPath. @sbalaji1192 can you confirm this? Can the widget's triggerPaths contain a primary column property path?
cc @ApekshaBhosale @rajatagrawal

Comment (vsvamsi1) [reply app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx:1]: Im not able to see any triggerPaths within a primaryColumn, Can you show me a screenshot of your app to be able to see it. Also do any of the triggerPaths within a primary column have evaluatedPopups associated to them.

Comment (vsvamsi1) [reply app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx:1]: @rishabhrathod01 Also if primaryColumns itself cannot be a triggerPath as what @sbalaji1192 mentioned,  i think we don't have to support as codeEditor does. @ApekshaBhosale is this okay?

Comment (sbalaji1192) [reply app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx:1]: @vsvamsi1 few column types have events , those are triggerPaths right?

Comment (ApekshaBhosale) [reply app/client/src/components/propertyControls/PrimaryColumnsControlV2.tsx:1]: yes this seems okay @vsvamsi1

File: app/client/src/ce/workers/Evaluation/dataTreeUtils.ts [hunk 1]
```diff
@@ -49,30 +49,12 @@ export function makeEntityConfigsAsObjProperties(
...
-  for (const [entityName, entityEvalProps] of Object.entries(
-    sanitizedEvalProps,
-  )) {
+  for (const [entityName, entityEvalProps] of Object.entries(evalProps)) {
     if (!entityEvalProps.__evaluation__) continue;
     set(
       dataTreeToReturn[entityName],
       "__evaluation__",
-      entityEvalProps.__evaluation__,
+      klona({ errors: entityEvalProps.__evaluation__.errors }),
```
Comment (rishabhrathod01) [inline app/client/src/ce/workers/Evaluation/dataTreeUtils.ts:57]: Do we need klona here, it wasn't present in the previous change.

**Review by rishabhrathod01 — APPROVED** [review]

**Review by rishabhrathod01 — APPROVED** [review]
Changes LGTM. 
Have added nit comment

File: app/client/src/ce/workers/Evaluation/dataTreeUtils.ts (hunk 1 above)
Comment (vsvamsi1) [reply app/client/src/ce/workers/Evaluation/dataTreeUtils.ts:57]: Yeah we have to scrub out functions from evalProps, that logic was there earlier (we were doing JSON.parse(JSON.Stringify()) but now we are doing the same through klona/json which is more faster.

**Pull request merged**